### PR TITLE
Add build github action

### PR DESCRIPTION
**Pull request by oliel**
Signed-off-by: Ori Liel <oliel@redhat.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -19,6 +19,70 @@ jobs:
 
     name: ${{ matrix.name }}
 
+    env:
+      ARTIFACTS_DIR: exported-artifacts
+
+    container:
+      image: quay.io/centos/centos:${{ matrix.container-name }}
```
Comment (mwperina) [inline .github/workflows/build.yml:26]: Please try to use oVirt build container:

https://github.com/oVirt/ovirt-engine-extensions-api/blob/master/.github/workflows/build.yml#L18
https://github.com/oVirt/ovirt-engine-extensions-api/blob/master/.github/workflows/build.yml#L29

 It should have installed all required dependencies as soon as you have RPM published to COPR:

https://github.com/oVirt/buildcontainer/blob/main/Dockerfile.stream8#L29

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -19,6 +19,70 @@ jobs:
...
+
     steps:
-      - name: Sanity Check
-        run: echo Hello world
+      - name: Prepare CentOS Stream 8 environment
```
Comment (mwperina) [inline .github/workflows/build.yml:29]: This step is not needed if you use oVirt buildcontainer

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -19,6 +19,70 @@ jobs:
...
+        if: ${{ matrix.shortcut == 'cs8' }}
+        run: |
+          # Install oVirt repositories
+          dnf copr enable -y ovirt/ovirt-master-snapshot centos-stream-8
+          dnf install -y ovirt-release-master
+          # Configure CS8 repositories
+          dnf config-manager --enable powertools
+          dnf module enable -y pki-deps javapackages-tools
+      - name: Install required packages
```
Comment (mwperina) [inline .github/workflows/build.yml:38]: Most of those dependencies should already be in if you use oVirt buildcontainer, please minimize the list

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -19,6 +19,70 @@ jobs:
...
+        uses: actions/checkout@v2
+
+      - name: Use cache for maven
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Perform build
```
Comment (mwperina) [inline .github/workflows/build.yml:73]: Please add empty line before step for readability

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -19,6 +19,70 @@ jobs:
...
+        run: |
+          python3 automation/build.py $ARTIFACTS_DIR
+      - name: Upload RPM artifacts
+        uses: actions/upload-artifact@v2
```
Comment (mwperina) [inline .github/workflows/build.yml:77]: Please use ovirt-upload-rpms action instead
https://github.com/oVirt/ovirt-engine-extensions-api/blob/master/.github/workflows/build.yml#L48

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -19,6 +19,70 @@ jobs:
...
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Perform build
+        run: |
+          python3 automation/build.py $ARTIFACTS_DIR
+      - name: Upload RPM artifacts
```
Comment (mwperina) [inline .github/workflows/build.yml:76]: Please add empty line before step for readability

**Review by mwperina — CHANGES_REQUESTED** [review]

**Comment by oliel** [comment]
Since we currently don't have a working build workflow and this PR adds such, I will first merge it and then implement above comments in a separate PR, as they are not trivial

**Pull request merged**